guides/source/getting_started.md

```diff
@@ -7,1150 +7,1551 @@ This guide covers getting up and running with Ruby on Rails.
-link's text and its second argument as the link's destination. If we pass a
-model object as the second argument, `link_to` will call the appropriate path
-helper to convert the object to a path. For example, if we pass an article,
-`link_to` will call `article_path`. So `app/views/articles/index.html.erb`
-becomes:
-
-```html+erb
-<h1>Articles</h1>
-
-<ul>
-  <% @articles.each do |article| %>
-    <li>
-      <%= link_to article.title, article %>
-    </li>
-  <% end %>
-</ul>
 ```
+http://example.org/products?sale=true&sort=asc
+```
+
+In this URL, each part has a name:
+
+- `https` is the **protocol**
+- `example.org` is the **host**
+- `/products` is the **path**
+- `?sale=true&sort=asc` are the **query parameters**
 
-Nice!
+### HTTP Methods and Their Purpose
 
-TIP: To learn more about routing, see [Rails Routing from the Outside In](
-routing.html).
+HTTP requests use methods to tell a server what action it should perform for a given URL. Here are the most common methods:
 
-### Creating a New Article
+- A `GET` request tells the server to retrieve the data for a given URL (e.g., loading a page or fetching a record).
+- A `POST` request will submit data to the URL for processing (usually creating a new record).
+- A `PUT` or `PATCH` request submits data to a URL to update an existing record.
+- A `DELETE` request to a URL tells the server to delete a record.
 
-Now we move on to the "C" (Create) of CRUD. Typically, in web applications,
-creating a new resource is a multi-step process. First, the user requests a form
-to fill out. Then, the user submits the form. If there are no errors, then the
-resource is created and some kind of confirmation is displayed. Else, the form
-is redisplayed with error messages, and the process is repeated.
+### Rails Routes
 
-In a Rails application, these steps are conventionally handled by a controller's
-`new` and `create` actions. Let's add a typical implementation of these actions
-to `app/controllers/articles_controller.rb`, below the `show` action:
+A `route` in Rails refers to a line of code that pairs an HTTP Method and a URL path. The route also tells Rails which `controller` and `action` should respond to a request.
+
+To define a route in Rails, let's go back to your code editor and add the following route to `config/routes.rb`
 
 ```ruby
-class ArticlesController < ApplicationController
-  def index
-    @articles = Article.all
-  end
+Rails.application.routes.draw do
+  get "/products", to: "products#index"
+end
+```
 
-  def show
-    @article = Article.find(params[:id])
-  end
+This route tells Rails to look for GET requests to the `/products` path. In this example, we specified `"products#index"` for where to route the request.
 
-  def new
-    @article = Article.new
-  end
+When Rails sees a request that matches, it will send the request to the `ProductsController` and the `index` action inside of that controller. This is how Rails will the request and return a response to the browser.
```

Comment:
```suggestion
When Rails sees a request that matches, it will send the request to the `ProductsController` and the `index` action inside of that controller. This is how Rails will map/process the request and return a response to the browser.
```